lib/CFRPageActions.jsm

```diff
@@ -275,6 +276,52 @@ class PageAction {
     }
   }
 
+  async _renderPinTabAnimation() {
+    const footer = this.window.document.getElementById("cfr-notification-footer");
+    const ANIMATION_CONTAINER_ID = "cfr-notification-footer-pintab-animation-container";
+    let animationsEnabled = Services.prefs.getBoolPref(ANIMATIONS_ENABLED_PREF, true);
+    let animationContainer = this.window.document.getElementById(ANIMATION_CONTAINER_ID);
+    if (animationContainer) {
+      animationContainer.toggleAttribute("animate", animationsEnabled);
+      animationContainer.removeAttribute("paused");
+    } else {
+      animationContainer = this.window.document.createElement("vbox");
+      animationContainer.setAttribute("id", ANIMATION_CONTAINER_ID);
+      animationContainer.toggleAttribute("animate", animationsEnabled);
+
+      let spacer = this.window.document.createElement("vbox");
+      spacer.setAttribute("flex", 1);
+      animationContainer.appendChild(spacer);
+
+      let controlsContainer = this.window.document.createElement("hbox");
+      controlsContainer.setAttribute("id", "cfr-notification-footer-animation-controls");
+
+      spacer = this.window.document.createElement("vbox");
+      spacer.setAttribute("flex", 1);
```

Comment:
Could this be replaced by `margins` in the m-c CSS?

Reply:
possibly? Would it auto-grow? But the m-c change is already in autoland so it would have to be a followup.

Reply:
Oh I see. You mean not using a spacer and just setting padding-top or margin-top on the container. Makes sense 👍